File: app/src/main/java/com/sapuglha/tictactoe/di/AppModule.java
package com.sapuglha.tictactoe.di;

import com.sapuglha.tictactoe.controller.GameStatusHandler;
import com.sapuglha.tictactoe.model.GameStatus;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {
    @Provides
    GameStatus provideGameStatus() {
        return new GameStatus();
    }

    @Provides
    GameStatusHandler provideGameStatusHandler() {
        return new GameStatusHandler(new GameStatus());
    }
}

File: app/src/main/java/com/sapuglha/tictactoe/di/AppModule.kt
package com.sapuglha.tictactoe.di

import com.sapuglha.tictactoe.controller.GameStatusHandler
import com.sapuglha.tictactoe.model.GameStatus

import dagger.Module
import dagger.Provides

@Module
class AppModule {
    @Provides
    internal fun provideGameStatus(): GameStatus {
        return GameStatus()
    }

    @Provides
    internal fun provideGameStatusHandler(gameStatus: GameStatus): GameStatusHandler {
        return GameStatusHandler(gameStatus)
    }
}